java/org/brotli/wrapper/enc/Encoder.java

```diff
@@ -22,18 +23,76 @@
   final ByteBuffer inputBuffer;
   boolean closed;
 
+  /**
+   * https://www.brotli.org/encode.html#aa6f
+   * See encode.h, typedef enum BrotliEncoderMode
+   */
+  public enum Mode {
+    /**
+     * Default compression mode.
+     * In this mode compressor does not know anything in advance about the properties of the input.
+     */
+    GENERIC(0),
+    /**
+     * Compression mode for UTF-8 formatted text input.
+     */
+    TEXT(1),
+    /**
+     * Compression mode used in WOFF 2.0.
+     */
+    FONT(2);
+
+    private final int value;
+
+    Mode(int value) {
+      this.value = value;
+    }
+
+    public int value() {
+      return value;
+    }
+
+    public static Mode of(int value) {
+      switch (value) {
+        case 0:
+          return GENERIC;
+        case 1:
+          return TEXT;
+        case 2:
+          return FONT;
+        default:
+          throw new IllegalArgumentException("Unknown mode: " + value);
+      }
+    }
+
+    public static Mode of(String name) {
```

Comment:
Given that default is to throw IAE, we could simply use built-in valueOf here.

Reply:
(or simply drop this method)

Reply:
Dropped it!